src/Symfony/Component/String/UnicodeString.php

```diff
@@ -97,11 +97,15 @@ public function endsWith($suffix): bool
         $form = null === $this->ignoreCase ? \Normalizer::NFD : \Normalizer::NFC;
         normalizer_is_normalized($suffix, $form) ?: $suffix = normalizer_normalize($suffix, $form);
 
-        if ('' === $suffix || false === $suffix || false === $i = $this->ignoreCase ? grapheme_strripos($this->string, $suffix) : grapheme_strrpos($this->string, $suffix)) {
+        if ('' === $suffix || false === $suffix) {
             return false;
         }
 
-        return grapheme_strlen($this->string) - grapheme_strlen($suffix) === $i;
+        if ($this->ignoreCase) {
+            return grapheme_strlen($this->string) - grapheme_strlen($suffix) === grapheme_strripos($this->string, $suffix);
+        }
+
+        return $suffix === grapheme_extract($this->string, \strlen($suffix), GRAPHEME_EXTR_MAXBYTES, -\strlen($suffix));
```

Comment:
I'm not sure counting in bytes works with combining characters eg in NFD, this should differentiate an "e" from an "é".
Can you please add a test for this case?

Reply:
added tests.

it works because of this behavior: https://3v4l.org/3CLS0

Reply:
perfect, thank you